rootfs/etc/nginx/lua/global_throttle.lua

```diff
@@ -61,6 +62,13 @@ local function get_namespaced_key_value(namespace, key_value)
   return namespace .. key_value
 end
 
+local function set_headers(limit, count_current, window_size)
+      ngx.header["Ratelimit-Remaining"] = string.format("%.0f",
+        (limit - count_current <= 0 ) and 0 or (limit - count_current))
+      ngx.header["Ratelimit-Limit"] = limit
+      ngx.header["Ratelimit-Reset"] = window_size
```

Comment:
According to the spec, this should not be the window size. It should be time remaining in the current window. Lua Global Throttle does not provide that information today. I don't know what is the best way forward. Maybe we should add that support to https://github.com/ElvinEfendi/lua-resty-global-throttle first.

I'd rather not set `Ratelimit-Reset` until we find a way to set it correctly.

Reply:
I'll look into adding support in the lua-resty-global-throttle project. I agree we should not set it, for now we could just include a header Ratelimit-Window and reflect the window size as such.

Reply:
Looks like a relatively straightforward change to add the reset time. I've attempted to expose the remaining_time variable [here](https://github.com/ElvinEfendi/lua-resty-memcached/compare/master...kaladaOpuiyo:main?expand=1) which appears to be useful in determining the reset time.

Reply:
Hello @ElvinEfendi I opened a pr against the lua-resty-global-throttle project to return the remaining_time, which I believe can be used to set the Ratelimit-Reset header. Please me review when you have a chance. [Expose remaining time as return value](https://github.com/ElvinEfendi/lua-resty-global-throttle/pull/29)